libs/api/domains/vehicles/src/lib/api-domains-vehicles.service.ts

```diff
@@ -0,0 +1,197 @@
+import { LOGGER_PROVIDER } from '@island.is/logging'
+import type { Logger } from '@island.is/logging'
+
+import { Inject, Injectable } from '@nestjs/common'
+import {
+  VehiclesApi,
+  BasicVehicleInformationGetRequest,
+  PersidnoLookupResult,
+  BasicVehicleInformationTechnicalMass,
+  BasicVehicleInformationTechnicalAxle,
+} from '@island.is/clients/vehicles'
+import { VehiclesAxle, VehiclesDetail } from '../models/getVehicleDetail.model'
+
+@Injectable()
+export class VehiclesService {
+  constructor(
+    @Inject(LOGGER_PROVIDER) private readonly logger: Logger,
+    @Inject(VehiclesApi)
+    private vehiclesApi: VehiclesApi,
+  ) {}
+
+  async getVehiclesForUser(nationalId: string): Promise<PersidnoLookupResult> {
+    const res = await this.vehiclesApi.vehicleHistoryGet({
+      requestedPersidno: nationalId,
+    })
+    return res
+  }
+
+  async getVehicleDetail(
+    input: BasicVehicleInformationGetRequest,
+  ): Promise<VehiclesDetail | null> {
+    const res = await this.vehiclesApi.basicVehicleInformationGet({
+      clientPersidno: input.clientPersidno,
+      permno: input.permno,
+      regno: input.regno,
+      vin: input.vin,
+    })
+    const { data } = res
+
+    if (!data) return null
+    const newestInspection = data.inspections?.sort((a, b) => {
+      if (a && b && a.date && b.date)
+        return new Date(a.date).getTime() - new Date(b.date).getTime()
+      else return 0
+    })[0]
+
+    const axleMaxWeight =
+      (data.techincal?.mass?.massdaxle1
```

Comment:
is this not a typo?

techincal vs. technical

Reply:
Yes it is - Samgöngustofa is going to fix in next api release